File: 2350-find-closest-number-to-zero/2350-find-closest-number-to-zero.py
class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        minimum = float('inf')
        res = 0

        for num in nums:
            if abs(num) < minimum:
                res = num
                minimum = abs(num)

            elif abs(num) == minimum:
                res = max(num, res)

        return res    

File: 2350-find-closest-number-to-zero/2350-find-closest-number-to-zero.go
func findClosestNumber(nums []int) int {
    minimum := math.MaxInt64
    res := 0
    for _, num := range nums {
        absNum := abs(num)

        if absNum < minimum {
            minimum = absNum
            res = num
        } else if absNum == minimum {
            res = max(num, res)
        } 
    }

    return res
}

func abs(x int) int {
    if x >= 0 {
        return x
    }

    return -x
}